client/my-sites/site-settings/controller.js

```diff
@@ -78,27 +77,18 @@ module.exports = {
 		}
 
 		// if user went directly to jetpack settings page, redirect
-		if ( site.jetpack && ! config.isEnabled( 'manage/jetpack' ) ) {
+		if ( site && site.jetpack && ! config.isEnabled( 'manage/jetpack' ) ) {
```

Comment:
Maybe use `isJetpackSite` and `getSelectedSiteId` here? Once we get rid of the `upgradeToBusiness` below, we will not need the entire `site` object and we can use more specific selectors. Also, you can use `canCurrentUser( state, siteId, 'manage_options' )` instead of `utils.userCan` that needs a `site` object.